.github/workflows/dvc-diff.yml

```
@@ -0,0 +1,95 @@
+# Checks for image diffs in a Pull Request and adds a GitHub comment showing the diff
+name: DVC image diff
+
+on:
+  pull_request:
+    paths:
+      - 'pygmt/tests/baseline/*.png.dvc'
+
+jobs:
+  dvc-diff:
+    name: DVC image diff
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2.3.4
+      with:
+        # fetch all history so that dvc diff works
+        fetch-depth: 0
+
+    - name: Setup data version control (DVC)
+      uses: iterative/setup-dvc@v1.0.2
+
+    - name: Setup continuous machine learning (CML)
+      uses: iterative/setup-cml@v1.0.0
+
+    # Produce the markdown diff report, which should look like:
+    # ## Summary of changed images
+    #
+    # This is an auto-generated report of images that have changed on the DVC remote
+    #
+    # | Status   | Path                                |
+    # |----------|-------------------------------------|
+    # | added    | pygmt/tests/baseline/test_image.png |
+    - name: Put list of images that were added or changed into report
+      run: |
+        echo -e "## Summary of changed images\n" > report.md
+        echo -e "This is an auto-generated report of images that have changed on the DVC remote\n" >> report.md
+        dvc diff --show-md master HEAD >> report.md
+        cat report.md
+
+    - name: Pull image data from cloud storage
+      run: dvc pull --remote upstream
+
+    - name: Put image diff(s) into report
+      env:
+        repo_token: ${{ secrets.GITHUB_TOKEN }}
+      id: image-diff
+      run: |
+        # Get just the filename of the changed image from the report
+        tail --lines=+7 report.md | awk '{print $4}' > diff_files.txt
```

This one is simpler. The command outputs the 4th columns (separated by spaces) after the 7th lines (NR is the current Number of Record).
```suggestion
        awk 'NR>=7 {print $4}' report.md > diff_files.txt
```